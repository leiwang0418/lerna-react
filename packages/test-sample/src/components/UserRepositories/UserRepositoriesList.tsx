import React, { FC } from 'react';
import PropTypes from 'prop-types';
import Typography from '@material-ui/core/Typography';
import List from '@material-ui/core/List';
import { useIntl, defineMessages } from 'react-intl';

interface Props {
	hasError?: boolean;
}

const messages = defineMessages({
	error: {
		defaultMessage: '加载用户仓库列表失败',
		description: '加载用户仓库列表错误',
	},
	test: {
		defaultMessage: '加载用户仓库列表失败',
		description: '加载用户仓库列表错误',
	},
});

const UserRepositoriesList: FC<Props> = ({ hasError }) => {
	const intl = useIntl();

	if (hasError) {
		return (
			<Typography variant="body1" color="error" gutterBottom>
				{intl.formatMessage(messages.error)}
			</Typography>
		);
	}

	return (
		<List>
			-----
			<Typography variant="body1" color="error" gutterBottom>
				{intl.formatMessage(messages.test)}
			</Typography>
		</List>
	);
};

UserRepositoriesList.propTypes = {
	hasError: PropTypes.bool,
};

export default UserRepositoriesList;
