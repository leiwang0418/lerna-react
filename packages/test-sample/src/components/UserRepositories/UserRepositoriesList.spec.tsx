import { render } from "@test-utils";
import UserRepositoriesList, {
	RepositoriesProps,
} from "./UserRepositoriesList";

jest.mock("./UserRepositoriesListItem", () => () => (
	<li>UserRepositoriesListItem</li>
));

describe("UserRepositoriesList", () => {
	it("should render error page", () => {
		const { container } = render(<UserRepositoriesList hasError={true} />);

		expect(container).toMatchSnapshot();
	});

	it("should render empy page", () => {
		const { container } = render(
			<UserRepositoriesList isFetching={false} />
		);

		expect(container).toMatchSnapshot();
	});

	it("should render UserRepositoriesList", () => {
		const props: RepositoriesProps = {
			isFetching: true,
			repositories: [
				{
					id: 1,
					name: "name",
					description: "description",
					html_url: "html_url",
				},
			],
		};

		const { container } = render(<UserRepositoriesList {...props} />);

		expect(container).toMatchSnapshot();
	});
});
