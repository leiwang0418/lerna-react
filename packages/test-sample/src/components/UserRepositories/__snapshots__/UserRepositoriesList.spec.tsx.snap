// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserRepositoriesList should render UserRepositoriesList 1`] = `
<div>
  <ul
    class="MuiList-root MuiList-padding"
  >
    <li>
      UserRepositoriesListItem
    </li>
  </ul>
</div>
`;

exports[`UserRepositoriesList should render empy page 1`] = `
<div>
  <ul
    class="MuiList-root MuiList-padding"
  >
    <li
      class="MuiListItem-root MuiListItem-gutters MuiListItem-alignItemsFlexStart"
    >
      <div
        class="MuiListItemText-root"
      >
        <span
          class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
        >
          未发现仓库数据
        </span>
      </div>
    </li>
  </ul>
</div>
`;

exports[`UserRepositoriesList should render error page 1`] = `
<div>
  <p
    class="MuiTypography-root MuiTypography-body1 MuiTypography-colorError MuiTypography-gutterBottom"
  >
    加载用户仓库列表失败
  </p>
</div>
`;
